1.0/docs/devguide/tests.md

```diff
@@ -0,0 +1,367 @@
+the implementation for all elements of a given type. 
+
+## Create stub elements
+
+Use `replace()` to create stub elements. 
+
+    setup(function() {
+      replace('paper-button').with('fake-paper-button');
+    });
+
+For example, using the sample `replace()` above and the element below:
+
+    <dom-module id='x-el'>
+      <template>
+        <paper-button id="pb">button</paper-button>
+      </template>
+    </dom-module>
+
+At test runtime, the content template would be stamped out as:
+
+    <dom-module id='x-el'>
+      <template>
+        <fake-paper-button id="pb">button</fake-paper-button>
+      </template>
+    </dom-module>
+
+The attributes and content of the element are preserved, but the tag
+is replaced with the specified stub tag.
+
+Because the method is called within `setup()`, all of the changes are 
+reverted at the end of each test.
+
+## AJAX
+
+<div class="yt-embed">
+  <google-youtube
+    videoid="_9qARcdCAn4"
+    autoplay="0"
+    rel="0"
+    fluid>
+  </google-youtube>
+</div>
+
+`wct` includes [Sinon](http://sinonjs.org/), which enables you to mock XHR
+requests and create fake servers. 
+
+Below is an example of a simple XHR unit test suite for 
+[`<iron-ajax>`](https://elements.polymer-project.org/elements/iron-ajax).
+Check out Sinon's documentation for more in-depth examples. 
+
+{% highlight html %}
+<!doctype html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" 
+        content="width=device-width, minimum-scale=1.0, initial-scale=1.0, user-scalable=yes">
+  <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+  <script src="../../web-component-tester/browser.js"></script>
+  <!-- import the element -->
+  <link rel="import" href="../bower_components/iron-ajax/iron-ajax.html">
+</head>
+<body>
+  <!-- create test fixture template -->
+  <test-fixture id="simple-get">
+    <template>
+      <iron-ajax url="/responds_to_get_with_json"></iron-ajax>
+    </template>
+  </test-fixture>
+  <script>
+    suite('<iron-ajax>', function() {
+      var ajax;
+      var request;
+      var server;
+      var responseHeaders = {
+        json: { 'Content-Type': 'application/json' }
+      };
+      setup(function() {
+        server = sinon.fakeServer.create();
+        server.respondWith(
+          'GET',
+          /\/responds_to_get_with_json.*/, [
+            200,
+            responseHeaders.json,
+            '{"success":true}'
+          ]
+        );
+      });
+      teardown(function() {
+        server.restore();
+      });
+      suite('when making simple GET requests for JSON', function() {
+        setup(function() {
+          // get fresh instance of iron-ajax before every test
+          ajax = fixture('simple-get');
+        });
+        test('has sane defaults that love you', function() {
+          request = ajax.generateRequest();
+          server.respond();
+          expect(request.response).to.be.ok;
```

Comment:
test switched to expect here, where previously the article was using assert. Might need to call out Chai somewhere?